ow_power_system/include/PowerSystemNode.h

```diff
@@ -11,121 +21,166 @@
+  ros::Publisher m_state_of_charge_pub;        // State of Charge Publisher
+  ros::Publisher m_remaining_useful_life_pub;  // Remaining Useful Life Publisher
+  ros::Publisher m_battery_temperature_pub;    // Battery Temperature Publisher
+  ros::Subscriber m_joint_states_sub;          // Mechanical Power Subscriber
+
+  // Flags that determine if debug output is printed regarding battery status
+  // during runtime.
+  // NOTE: There's no function to get boolean variables from a config, so
+  // these bools have to be set via string comparisons. Could benefit from an
+  // update if such a function is added in the future.
+  bool m_print_debug = false;
+  bool m_timestamp_print_debug = false;
+  bool m_inputs_print_debug = false;
+  bool m_outputs_print_debug = false;
+  bool m_topics_print_debug = false;
+  bool m_mech_power_print_debug = false;
+
+  // system.cfg variables:
 
   // HACK ALERT.  The prognoser produced erratic/erroneous output when
-  // given too high a power input.  This made-up value protects
-  // against this, but is a temporary hack until a circuit breaker
-  // model is added to the power system, and/or the multi-pack battery
-  // model is implemented and can handle any envisioned power draw.
-  // This initial value is overriden by the system config.
-  //
-  double m_max_gsap_input_watts = 30;
-
-  // Number of lines in power fault profiles to skip, in order to
-  // synchonize the consumption of the profile with GSAP's cycle rate
-  // (above).  This computation is left to the user for now, though
-  // note that the default values are not expected to change as of
-  // Release 9.  This initial value is overriden by the system config.
-  int m_profile_increment = 2;
+  // given too high a power input.  The value assigned to this in system.cfg
+  // protects against this by capping the power input, but it is a temporary
+  // hack until a circuit breaker model is added to the power system and/or 
+  // the multi-cell battery model is implemented and can handle any envisioned
+  // power draw.
+  double m_max_gsap_input_watts;
+
+  // The maximum RUL estimation output from the Monte Carlo prediction process.
+  // If the prediction hits this value, it stops immediately and returns infinity
+  // (which is processed later into the max value instead).
+  // Lower values mean faster performance in the event a RUL prediction would
+  // exceed the horizon value, but it also means the prognoser can't return RUL
+  // values higher than this.
+  int m_max_horizon_secs;
+
+  // The number of samples each model creates during the Monte Carlo prediction
+  // process. Lower values mean faster performance, but lower accuracy (needs
+  // testing for confirmation).
+  int m_num_samples;
 
   // Number of threads to use for the asynchronous ROS spinning in the main
   // loop (to allow jointStatesCb to run at 50Hz separate from the 0.5Hz main
   // loop).
-  int m_spinner_threads = 4;
+  int m_spinner_threads;
```

Comment:
Yes, these spinner threads are just for ROS callback functions so we can call ``jointStatesCb`` mid-cycle unlike before. ``4`` was just the default value and we can change it if needed, which is why I put its initialization into ``system.cfg``.

Reply:
I suspect it could be done with fewer threads, but seeing as 4 is the default, I think that is a reasonable value to set here barring exhaustive testing.  👍🏻